app/command_autocomplete_test.go

```diff
@@ -132,71 +132,166 @@ func TestParseInputTextArgument(t *testing.T) {
 	assert.False(t, found)
 	assert.Equal(t, "--name ", parsed)
 	assert.Equal(t, "bla", toBeParsed)
 
-	found, parsed, toBeParsed, _ = parseNamedArgument(argument, "", "--name bla gla")
+	found, parsed, toBeParsed, _ = parseNamedArgument(argumentTypeText, "", "--name bla gla")
 	assert.False(t, found)
 	assert.Equal(t, "--name ", parsed)
 	assert.Equal(t, "bla gla", toBeParsed)
 
-	found, parsed, toBeParsed, _ = parseNamedArgument(argument, "", "--name \"bla gla\"")
+	found, parsed, toBeParsed, _ = parseNamedArgument(argumentTypeText, "", "--name \"bla gla\"")
 	assert.False(t, found)
 	assert.Equal(t, "--name ", parsed)
 	assert.Equal(t, "\"bla gla\"", toBeParsed)
 
-	found, parsed, toBeParsed, _ = parseNamedArgument(argument, "", "--name \"bla gla\" ")
+	found, parsed, toBeParsed, _ = parseNamedArgument(argumentTypeText, "", "--name \"bla gla\" ")
 	assert.False(t, found)
 	assert.Equal(t, "--name ", parsed)
 	assert.Equal(t, "\"bla gla\" ", toBeParsed)
 
-	found, parsed, toBeParsed, _ = parseNamedArgument(argument, "", "bla")
+	found, parsed, toBeParsed, _ = parseNamedArgument(argumentTypeBool, "", "bla")
 	assert.False(t, found)
 	assert.Equal(t, "bla", parsed)
 	assert.Equal(t, "", toBeParsed)
 
 }
 
+func TestParseBoolArgument(t *testing.T) {
+	argument := &model.AutocompleteArg{
+		Name:     "", //positional
+		HelpText: "some_help",
+		Type:     model.AutocompleteArgTypeText,
+		Data:     &model.AutocompleteBoolArg{Hint: "hint", BoolValue: "BoolValue"},
+	}
+
+	found, _, _, suggestion := parseBoolArgument(argument, "", "")
+	assert.True(t, found)
+	assert.Equal(t, model.AutocompleteSuggestion{Complete: "", Suggestion: "", Hint: "hint", Description: "some_help"}, suggestion)
+
+	found, _, _, suggestion = parseBoolArgument(argument, "", " ")
+	assert.True(t, found)
+	assert.Equal(t, model.AutocompleteSuggestion{Complete: " ", Suggestion: "", Hint: "hint", Description: "some_help"}, suggestion)
+
+	found, _, _, suggestion = parseBoolArgument(argument, "", "tru")
+	assert.True(t, found)
+	assert.Equal(t, model.AutocompleteSuggestion{Complete: "true", Suggestion: "", Hint: "hint", Description: "some_help"}, suggestion)
+
+	found, _, _, suggestion = parseBoolArgument(argument, "", "false")
+	assert.True(t, found)
+	assert.Equal(t, model.AutocompleteSuggestion{Complete: "false", Suggestion: "", Hint: "hint", Description: "some_help"}, suggestion)
+
+	found, parsed, toBeParsed, _ := parseBoolArgument(argument, "", "true test_command")
+	assert.False(t, found)
+	assert.Equal(t, "true ", parsed)
+	assert.Equal(t, "test_command", toBeParsed)
+
+	found, parsed, toBeParsed, _ = parseBoolArgument(argument, "", "false test_command")
+	assert.False(t, found)
+	assert.Equal(t, "false ", parsed)
+	assert.Equal(t, "test_command", toBeParsed)
+
+	found, parsed, toBeParsed, _ = parseBoolArgument(argument, "", "abc ")
+	assert.False(t, found)
+	assert.Equal(t, "abc ", parsed)
+	assert.Equal(t, "", toBeParsed)
+}
+
 func TestSuggestions(t *testing.T) {
```

Comment:
Could we add some unit tests here? In the `TestCommandWithOptionalArgs` function as well. Let's test bool arguments in the complex environment with different type of args, including optional args.

Reply:
sure i will add that